default/python/AI/AIstate.py

```diff
@@ -74,7 +74,12 @@ def convert_to_version(state, version):
 
     if version == 0:
```

Comment:
> As I see you want to write this by hand.

No. That was never the intention.

The two possible choices are:
1. A fall-through pseudo-"switch"-statement:
```
if state.version == a:
    a_b(state)  # state.version is now b
if state.version == b:  # also is true if initial version was a
    b_c(state)  # state.version is now c
```

2. elif-statements as in this PR and iteratively convert the state by calling the function with increasing version number in a for-loop, i.e. upgrade only by 1 version by function call (cleaner than the fall-through switch if additional error checks are implemented that should be done between each step).

In any case, the ```b_c()``` part is written only once here:
```
(el)if version == b():
    b_c()
```